_posts/2020-08-26-gsoc-webgpu.md

```diff
@@ -0,0 +1,89 @@
+---
+layout:     post
+title:      GSoC wrap-up - Implementing WebGPU in Servo
+date:       2020-08-26 00:30:00
+summary:    A walk through the implementation of WebGPU in Servo
+categories: 
+---
+
+##Introduction
+Hello everyone! I am Kunal([@kunalmohan](https://github.com/kunalmohan)), an undergrad student at Indian Institute of Technology Roorkee, India. As a part of Google Summer of Code(GSoC) 2020, I worked on implementing WebGPU in Servo under the mentorship of Mr. Dzmitry Malyshau([@kvark](https://github.com/kvark)). I devoted the past 3 months working on ways to bring the API to fruition in Servo, so that Servo is able to run the existing examples and pass the Conformance Test Suite(CTS). This is going to be a brief account of how I started with the project, what challenges I faced and how I overcame them.
+
+##What is WebGPU?
+WebGPU is a future web standard, cross-platform graphics API aimed to make GPU capabilities more accessible on the web. WebGPU is designed from the ground up to efficiently map to the Vulkan, Direct3D 12, and Metal native GPU APIs. A native implementation of the API in Rust is present in the [wgpu crate](https://github.com/gfx-rs/wgpu). Servo implementation of the API uses this crate.
+
+##The Project
+At the start of the project the implementation was in a pretty raw state- Servo was only able to accept shaders as SPIRV binary and ran just the compute example. I had the following tasks in front of me-
+- Implement the various [DOM interfaces](https://gpuweb.github.io/gpuweb/#idl-index) that build up the API
+- Setup a proper Id rotation for the GPU resources.
+- Integrate WebGPU with WebRender for rendering the textures on HTML canvas.
+- Setup proper model model for async error recording.
+
+The final goal was to be able to run the live examples at https://austineng.github.io/webgpu-samples/ and pass a fair amount of the CTS.
```

Comment:
since this is the first time CTS is mentioned, let's clarify what this is

Reply:
I have mentioned CTS under `Introduction` before this(with the full form) and provided a one-line description under the separate heading.